crates/host/src/module/wasmer_wamr.rs

```diff
@@ -15,12 +15,15 @@ pub fn make_runtime_engine() -> Engine {
     Engine::default()
 }
 
-/// Take WASM binary and prepare a wasmer Module suitable for iOS
-pub fn build_ios_module(_wasm: &[u8]) -> Result<Module, CompileError> {
+/// Compile a wasm binary, serialize it with wasmer's serializtion format, and write to a file.
+/// This file can later be used for contexts where JIT compilation is not possible (iOS for example).
+pub fn write_precompiled_serialized_module_to_file(wasm: &[u8], path: PathBuf) -> Result<(), CompileError> {
     unimplemented!("The feature flag 'wasmer_sys' must be enabled to support compiling wasm");
 }
 
-/// Deserialize a previously compiled module for iOS from a file.
-pub fn get_ios_module_from_file(_path: &Path) -> Result<Module, DeserializeError> {
+/// Deserialize a previously precompiled and serialized module. 
+/// Even though the `wasmer_wamr` feature flag supports deserializing a pre-compiled and serialized module,
+/// it doesn't make sense to use a pre-compiled module as it would be executed by the interpreter engine anyway.
+pub fn read_precompiled_serialized_module_from_file(path: &Path) -> Result<Module, DeserializeError> {
```

Comment:
Not sure if this should share the same implementation as wasmer_sys, or keep this panic.